WCFDataService/Client/System/Data/Services/Client/DataServiceRequest.cs

```diff
@@ -272,6 +273,10 @@ internal long GetQuerySetCount(DataServiceContext context)
 
                 throw;
             }
+            finally
+            {
```

Comment:
No, no enumeration occurs for GetQuerySetCount, it simply returns a long with count value returned in the message. The message body is read entirely within a stream reader, like:  r = XmlConvert.ToInt64(sr.ReadToEnd()); We are done with the message after this and it is safe to dispose it off.